### ISSUE TITLE
Feature request: avoid limits of integration

### ISSUE THREAD

**Issue by quantumAnsatz** [saullocastro/cubature#31]
When integrating it is often desired to avoid the limits of integration as some functions may be ill-defined when evaluated at the endpoints.

Here is a minimal example

```
from cubature import cubature
from math import sin,pi

def integrand(x):
    out=np.sin(pi*x)/(x*pi)
    return out

x0=0.25937986 # unimportant example of upper boundary

cubature(integrand,1,1,[0],[x0],abserr=0,relerr=1.e-3,vectorized=False,adaptive='h')

```
with the following result:
```
/var/folders/9m/sy48svs95j54p3wr659d_w8r0000gn/T/ipykernel_3010/3647069318.py:6: RuntimeWarning: invalid value encountered in true_divide

out=np.sin(pi*x)/(x*pi)

(array([0.25]), array([2.77555755e-15]))
```
The result is correct, however along with the result, it provides a warning as cubature is evaluating the integrand at the lower bound`x=0`. For the example given here, it is harmless. However for more complicated functions, evaluating the function at the boundaries can produce significant errors and/or make the calculation to never halt, specially when dealing with higher dimensions.

There are ways to bypass this problem, but It would be nice to have the option to automatically avoid endpoints. As far as I know avoiding the limits of integration is the default for h-adaptive in cubature C package.

**Comment by saullocastro** [comment]
@quantumAnsatz thank you for the feature request

Would it be enough to use a solution whereby the user specifies to avoid the integration limits? Internally, the integration function could change the limits from `x1, x2` to `x1+tol, x2-tol`. What do you think?

A possible solution is to add another boolean parameter such as:

`cubature(integrand, 1, 1, [0], [x0], abserr=0, relerr=1.e-3, vectorized=False, adaptive='h', avoid_integration_limits=True)`

**Comment by quantumAnsatz** [comment]
@saullocastro thanks to you for this amazing module.

I think I have found the issue. For me it was uncanny that cubature h-adaptive was evaluating the functions in the limits of integration, see [Gauss–Kronrod quadrature formula](https://en.wikipedia.org/wiki/Gauss%E2%80%93Kronrod_quadrature_formula) and the README.md in the C version of cubature. 

However there was a systematic check before using the cython module to call for cubature. Using python, the dimensions of the integrals were verified by evaluating the function at the lower bound `xmin`.

I have created a pull request that should bypass this issue. 
Using single and multi-variable integrations, I have tested that it works and that now the functions is never evaluated at `xmin`. Please do not hesitate to double check.

Best regards.

**Comment by saullocastro** [comment]
@quantumAnsatz thank you for the contribution! I have released version 0.15.0 already incorporating this bug fix.
